### DISCUSSION TITLE
v7 Column TS type is defective ('disableSortBy' does not exist in type 'Column<{}>)

### DISCUSSION THREAD

**Discussion by tomliang-amzn** [TanStack/table#4400] [General] [answered]
react-table version: 7.8.0
@types/react-table version: 7.7.12
A Codesandbox link to reproduce the issues: https://codesandbox.io/s/react-table-disablesortby-typescript-h4wuhd?file=/src/App.tsx

If I create the columns object with `Column` type, TS is giving a lot of false alarms.

For example, if we want to disable sort on a particular column, we can pass `disabledSortBy: false` to the column object. This can be seen at https://github.com/TanStack/table/issues/605. However, TS would complain about it:
> 'disableSortBy' does not exist in type 'Column<{}>'

I wonder if I am supposed to use other types for columns or the `Column` type is defective ?

**Comment by tomliang-amzn** [comment, marked as answer]
I figured it out.

Long answer: https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/react-table

tldr:
installing the type library `@types/react-table` doesn't mean we are all set with types for react-table. We need to create a file named `react-table-config.d.ts` in order for types depend upon declaration merging to work well. So depending on what what react-table plugin hooks you want to include, you can play around with the types in `react-table-config.d.ts` to make it give more or less type suggestion for the column or table.
You may also need to change `tsconfig.json` to include `react-table-config.d.ts`.
